boa/src/value/mod.rs

```diff
@@ -465,11 +466,14 @@ impl Value {
     /// Set_prop, which will overwrite prop with a new Property
     ///
     /// Mostly used internally for now
-    pub(crate) fn update_property(&self, field: &str, new_property: Property) {
+    pub(crate) fn update_property<P>(&self, field: &str, new_property: P)
+    where
+        P: Into<PropertyDescriptor>,
+    {
         let _timer = BoaProfiler::global().start_event("Value::update_property", "value");
 
         if let Some(ref mut object) = self.as_object_mut() {
-            object.insert(field, new_property);
+            object.insert(field, new_property.into());
         }
     }
```

Comment:
Since we're touching this code might as well change the type of `field` to be `K: Into<PropertyKey>`.

Reply:
> Since we're touching this code might as well change the type of `field` to be `K: Into<PropertyKey>`.

I think its best if I remove this function its used in a very small amount of paces, and we should be slowly removing the Value object behaviour #577